modin/core/dataframe/pandas/partitioning/partition.py

```diff
@@ -298,10 +298,7 @@ def length(self):
             The length of the object.
         """
         if self._length_cache is None:
-            cls = type(self)
-            func = cls._length_extraction_fn()
-            preprocessed_func = cls.preprocess_func(func)
-            self._length_cache = self.apply(preprocessed_func)
+            self._length_cache = self.apply(self._length_extraction_fn()).get()
```

Comment:
Ha, this is interesting. Doesn't `self._length_extraction_fn()` effectively do `type(self)._length_extraction_fn()`?

Reply:
Almost, applying to an object, not a class, the search space should be slightly larger, but this does not bother us in this case. Therefore, I do not think that we need to do an additional action to calculate the class.